stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/CheckboxFieldUI.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Info
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.R
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.stateDescription
+import androidx.compose.ui.semantics.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.uicore.stripeColors
+
+@Composable
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun CheckboxFieldUI(
+    controller: CheckboxFieldController,
+    enabled: Boolean = true
+) {
+    val isChecked = controller.isChecked.collectAsState().value
+    val error = controller.error.collectAsState(initial = null).value
+
+    CheckboxFieldUIView(
+        isChecked = isChecked,
+        enabled = enabled,
+        debugTag = controller.debugTag,
+        onValueChange = controller::onValueChange,
+        label = @ReadOnlyComposable {
+            controller.labelResource?.let { resource ->
+                stringResource(id = resource.labelId, formatArgs = resource.formatArgs)
+            } ?: ""
+        },
+        error = error?.let { fieldError ->
+            @ReadOnlyComposable {
+                fieldError.formatArgs?.let { args ->
+                    stringResource(id = fieldError.errorMessage, formatArgs = args)
+                } ?: stringResource(id = fieldError.errorMessage)
+            }
+        }
+    )
+}
+
+@Composable
+internal fun CheckboxFieldUIView(
+    isChecked: Boolean,
+    enabled: Boolean,
+    debugTag: String,
+    onValueChange: (value: Boolean) -> Unit,
+    label: @Composable () -> String,
+    error: (@Composable () -> String)?,
+) {
+    val accessibilityDescription = stringResource(
+        if (isChecked) {
+            R.string.selected
+        } else {
+            R.string.not_selected
+        }
+    )
+
+    val errorColor = MaterialTheme.stripeColors.materialColors.error
+
+    val checkboxColors = error?.run {
+        CheckboxDefaults.colors(
+            checkedColor = errorColor,
+            uncheckedColor = errorColor,
+            checkmarkColor = errorColor
+        )
+    } ?: run {
+        CheckboxDefaults.colors(
+            checkedColor = MaterialTheme.colors.primary,
+            uncheckedColor = MaterialTheme.stripeColors.subtitle,
+            checkmarkColor = MaterialTheme.colors.surface
+        )
+    }
+
+    Column(
+        modifier = Modifier
+            .padding(vertical = 8.dp)
```

Comment:
We should try to have the padding handed _into_ the composables via the `modifier`, as opposed to them having their own internal padding.